packages/playwright-core/src/server/chromium/chromium.ts

```diff
@@ -349,6 +350,28 @@ export class Chromium extends BrowserType {
     chromeArguments.push(...args);
     return chromeArguments;
   }
+
+  override readyState(options: types.LaunchOptions): BrowserReadyState | undefined {
+    if (options.useWebSocket || options.args?.some(a => a.startsWith('--remote-debugging-port')))
+      return new ChromiumReadyState();
+    return undefined;
+  }
+}
+
+class ChromiumReadyState implements BrowserReadyState {
+  private readonly _wsEndpoint = new ManualPromise<string>();
+
+  onBrowserOutput(message: string): void {
+    const match = message.match(/DevTools listening on (.*)/);
+    if (match)
+      this._wsEndpoint.resolve(match[1]);
+  }
+  onBrowserExit(): void {
+    this._wsEndpoint.reject(new Error('Browser exited'));
```

Comment:
I wonder why does this one reject, and Juggler's one resolves?

Reply:
Let's have both reject.